docs/parameters.md

```diff
@@ -0,0 +1,185 @@
+# Passing parameters to brokers
+
+Table of Contents
+- [Overview](#overview)
+- [Design](#design)
+  - [Basic example](#basic-example)
+  - [Passing sensitive data](#passing-sensitive-data)
+    - [Reference to a secret](#reference-to-a-secret)
+    - [The entire payload with sensitive data](#the-entire-payload-with-sensitive-data)
+  - [Merging multiple sources and conflict resolution](#merging-multiple-sources-and-conflict-resolution)
+- [Example with multiple sources](#example-with-multiple-sources)
+
+## Overview
+`parameters` and `parametersFrom` properties of `Instance` and `Broker` resources 
+provide support for passing parameters to the broker relevant to the corresponding
+[provisioning](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#provisioning) or
+[binding](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#binding) request. 
+The resulting structure represents an arbitrary JSON object, which is assumed to 
+be valid for a particular broker. 
+The Service Catalog does not enforce any extra limitations on the format and content 
+of this structure.
+
+## Design
+
+When you create an `Instance` or a `Binding`, you can set parameters to be passed 
+to the corresponding broker.
+To set parameters, include the `parameters` or `parametersFrom` field in the spec.
+
+### Basic example
+
+Let's say we want to create an `Instance` of EC2 running on AWS using a
+[corresponding broker](https://github.com/cloudfoundry-samples/go_service_broker) 
+which implements the Open Service Broker API.
+
+A typical provisioning request for this broker looks [like this](https://github.com/cloudfoundry-samples/go_service_broker/blob/master/bin/curl_broker.sh):
+```bash
+curl -X PUT http://username:password@localhost:8001/v2/service_instances/instance_guid-111 -d '{
+  "service_id":"service-guid-111",
+  "plan_id":"plan-guid",
+  "organization_guid": "org-guid",
+  "space_guid":"space-guid",
+  "parameters": {"ami_id":"ami-ecb68a84"}
+}' -H "Content-Type: application/json"
+```
+
+Note that the broker accepts an `ami_id` parameter ([AMI](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) 
+identifier).
+To configure a provisioning request in Service Catalog, we need to declare an `Instance` 
+resource with an AMI identifier declared in the `parameters` field of its spec:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: ami-instance
+  namespace: test-ns
+spec:
+  serviceClassName: aws-ami
+  planName: default
+  parameters:
+    ami_id: ami-ecb68a84
+```
+
+### Passing sensitive data
+
+To pass parameters stored in a `Secret` resource, use the `parametersFrom` field.
```

Comment:
I think this above sentence might be misleading. I read it as "if you want to pass a single name/value pair that's in a secret, then use the `parametersFrom` field" - mainly because of what I just read in the previous section.  But I think what you really want to say is something more like:
- - - 
`Secret` resources can be used to store sensitive data. The following sections describe the two ways in which secrets can be used to populate the parameters sent to a broker.
- - -
Not saying whether we should keep support for the map[string]string variant or not, just saying that if we keep it I think this sentence needs to be tweaked.